src/main/kotlin/no/nav/emottak/App.kt

```diff
@@ -14,14 +14,16 @@ import no.nav.emottak.edi.adapter.plugin.configureCallLogging
 import no.nav.emottak.edi.adapter.plugin.configureContentNegotiation
 import no.nav.emottak.edi.adapter.plugin.configureMetrics
 import no.nav.emottak.edi.adapter.plugin.configureRoutes
+import no.nav.emottak.edi.adapter.util.DpopJwtProvider
 import org.slf4j.LoggerFactory
 
 internal val log = LoggerFactory.getLogger("no.nav.emottak.edi.adapter")
 
 fun main() = SuspendApp {
     result {
         resourceScope {
-            val deps = dependencies()
+            val dpopJwtProvider = DpopJwtProvider(config())
+            val deps = dependencies(dpopJwtProvider)
```

Comment:
Spørsmål: Gir det ikke mer mening å instansiere dpopJwtProvider dependency inni dependencies funksjonen?

Reply:
Jeg tror ikke det :) Har tenkt det gjennom altså. Provideren er en helper / utility som dependencies er avhenging av for å wire disse sammen korrekt. Det jeg liker med å han den utenfor og passe inn eksplisitt er at vi er tydelige på nettopp dette. 

I utgangspunktet ønsker vi én instans av denne. Og skulle det være behov for denne lengre ned i kjeden passer vi den bare videre fra toppnivå.

Reply:
Ombestemte meg 😄 Hadde allerede en annen util inne i dependencies() og strengt tatt skal bare Provideren brukes til wiring av dependencies.